tutorials/tutorial-howto-monitor-django-prometheus.md

```diff
@@ -0,0 +1,230 @@
+> - [Step 1 - Export prometheus-style monitoring metrics from your
+Django application](#step1)
+> - [Step 2 - Point Prometheus to your Django application metrics endpoint](#step2)
+> - [Step 3 - Instrument additional aspects of your application (optional)](#step3)
+> - [Next Steps](#next-steps)
+
+### Introduction [](introduction)
+Prometheus is an open-source systems monitoring and alerting toolkit that can be
+used to easily and cheaply monitor infrastructure and applications. In this
+tutorial we show how to monitor a Django application with Prometheus. (And even
+if you don’t have a Django application, we include an optional step to create
+one so that everyone can follow along.)
+
+### Prerequisites [](prereqs)
+A machine with the following installed:
+
+- Python
+- [pip][get-pip]
+- A locally running [Prometheus][get-prometheus] instance
+
+>:Tip: Note: Since machines commonly have multiple versions of Python installed, 
+in this tutorial we will call pip using the `python -m pip [foo]` syntax instead 
+of the `pip [foo]` syntax. This is to ensure that pip installs new components for
+the version of Python that we are using.
+
+### Step 0 - Set up a basic Django application (optional) [](step0)
+*(Please skip this step if you already have a Django application.)*
+
+#### Install [Django][get-django]
+```bash
+python -m pip install Django
+```
+
+#### Create a template project
+```bash
+cd [to the directory where you want to create the project]
+django-admin startproject mysite
+```
+
+This will create a mysite directory in your current directory, that looks
+something like this ([more here][django-first-app]):
+
+```
+mysite/
+    manage.py
+    mysite/
+        __init__.py
+        settings.py
+        urls.py
+        asgi.py
+        wsgi.py
+```
+
+#### Verify that Django is working
+Change to the outer `mysite` directory, and run:
+$ python manage.py runserver
+
+You should see something like this:
+
+````
+Performing system checks...
+
+System check identified no issues (0 silenced).
+
+You have unapplied migrations; your app may not work properly until they are applied.
+Run 'python manage.py migrate' to apply them.
+
+March 10, 2020 - 15:50:53
+Django version 3.0, using settings 'mysite.settings'
+Starting development server at http://127.0.0.1:8000/
+Quit the server with CONTROL-C.
+````
+
+If that works, then visit `http://127.0.0.1:8000/` from your Web browser.
+You should see a “Congratulations!” page, with a rocket taking off.
+
+(If that didn’t work, then please take a look at the
+[Django documentation][django-first-app] for troubleshooting.)
+
+### Step 1 - Export prometheus-style monitoring metrics from your Django application [](step1)
+
+We will use the [django-prometheus][get-django-prometheus] package for
+exporting prometheus-style monitoring metrics from our Django application.
+
+#### Install django-prometheus
+```bash
+python -m pip install django-prometheus
+```
+
+#### Modify settings.py
```

Comment:
```suggestion
#### Modify `settings.py`
```
This step and the next step can probably be combined under a 'Modify Django settings' or 'Configure Django', but if you want to keep the filename in the headline, use inline code